hail/src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -429,7 +428,9 @@ object Simplify {
       TableMapGlobals(TableHead(child, n), newGlobals)
 
     case TableHead(TableOrderBy(child, sortFields), n)
-      if sortFields.forall(_.sortOrder == Ascending) && n < 256 && canRepartition =>
+      if sortFields.forall(_.sortOrder == Ascending)
+        && child.typ.key != sortFields.map(_.field)
```

Comment:
This is too strict.  It should match the condition in table order by: that the sort fields are an prefix of the key.

Reply:
Which maybe you should break out as a separate function on object TableOrderBy and call in both places.